packages/financial-templates-lib/src/clients/ExpiringMultiPartyEventClient.js

```diff
@@ -40,6 +49,22 @@ class ExpiringMultiPartyEventClient {
     // Last block number to end the searching for events at.
     this.lastBlockToSearchUntil = endingBlockNumber;
     this.lastUpdateTimestamp = 0;
+
+    if (contractType !== "ExpiringMultiParty" && contractType !== "Perpetual")
+      throw new Error(
+        `Invalid contract type provided: ${contractType}! The financial product client only supports ExpiringMultiParty or Perpetual`
+      );
+    this.contractType = contractType;
+    if (
```

Comment:
note: we will improve this in the future with a better version comparison logic.